test/dates/periods.jl

```diff
@@ -292,6 +322,15 @@ dt = Dates.DateTime(2014)
 @test Dates.toms(d)  == Dates.value(Dates.Millisecond(d)) == 86400000
 @test Dates.toms(w)  == Dates.value(Dates.Millisecond(w)) == 604800000
 
+@test Dates.tons(ns) == Dates.value(Dates.Nanosecond(ns)) == 1
+@test Dates.tons(us) == Dates.value(Dates.Nanosecond(us)) == 1000
+@test Dates.tons(ms) == Dates.value(Dates.Nanosecond(ms)) == 1000000
+@test Dates.tons(s)  == Dates.value(Dates.Nanosecond(s))  == 1000000000
+@test Dates.tons(mi) == Dates.value(Dates.Nanosecond(mi)) == Int64(60000000000)
```

Comment:
I'm not at all familiar with this code, not sure I could be of much help. I don't know whether Gallium works on 32 bit Linux, but you could try the poor-man's debugger of following the dispatch trail with `@which`. CI is still running here, but I assume the issue is happening on both 32 bit Linux and 32 bit Windows?